README_fuzz_testing.md

```diff
@@ -0,0 +1,98 @@
+# Fuzz testing
+Halide has a set of fuzz-testing harnesses that can be used to find those
+tricky to find, edge cases and bugs that would otherwise not be caught
+by a regular unit-testing suite. At the moment these fuzz-tests are housed
+in the `//test/fuzz` directory. The fuzz testing suite use the common,
+[libfuzzer](https://www.llvm.org/docs/LibFuzzer.html) interface for fuzz-tests.
+
+## Building fuzz tests
+Fuzz testing requires specific instrumentation across the entire build,
+to do this we make use of a fuzzing-specific-toolchain/preset. e.g.
+
+```
+cmake . --preset linux-x64-fuzzer
+cmake --build ./build -j$(nproc)
+```
+
+## Using the fuzz-harnesses
+Fuzz-testing harnesses are a little different to a more traditional unit-test
+and would don't have a definitive end of test. In other words a fuzz test will
+run;
+- for an infinite amount of time (the default),
+- for a user specified maximum amount of time,
+- until the fuzzer finds a bug and crashes,
+- you manually kill the process e.g. (ctrl-C).
+
+Once you have built the fuzz testing suite using the commands listed above you
+can list the fuzz testing harnesses using the command;
+
+```
+ls ./build/test/fuzz/fuzz_*
+```
+
+To run a fuzzer simply run the fuzz-testing harness with no arguments. e.g.
+
+`./build/test/fuzz/fuzz_simplify`
+
+By default this will run the fuzz test on a single-core and discard whatever.
```

Comment:
```suggestion
By default this will run the fuzz test on a single core and discard whatever.
```
grammar nit: "single-core" with the hyphen is an adjective, but "single core" is just one core.